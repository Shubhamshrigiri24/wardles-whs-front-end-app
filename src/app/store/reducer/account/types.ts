import { TActionStatus } from "app/store/types";
import { ACCOUNT_ACTIONS } from "app/store/reducer/account/constants";

export type TAccount = {
  dob: string;
  email: string;
  firstName: string;
  gender: string;
  lastName: string;
  phone: string;
  updateTime: string;
  userId: string;
  addressline1: string;
  addressline2: string;
  agreeMarketing: boolean;
  city: string;
  gpDetails: string;
  postcode: string;
};

export type State = {
  data: TAccount | null;
  status: TActionStatus;
};

export type SetAccountDataActionType = {
  type: typeof ACCOUNT_ACTIONS.SET_ACCOUNT;
  payload: TAccount;
};

export type SetAccountStatusActionType = {
  type: typeof ACCOUNT_ACTIONS.SET_ACCOUNT_STATUS;
  payload: TActionStatus;
};

export type AccountActionTypes =
  | SetAccountDataActionType
  | SetAccountStatusActionType;
