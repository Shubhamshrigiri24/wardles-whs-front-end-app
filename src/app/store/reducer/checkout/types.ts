import { RSAA } from "redux-api-middleware";
import { PostcodeLookupAddress } from "../shared";

export interface State {
  customerDetails: CustomerDetails;
  confirmedPayment: boolean;
  braintreeToken: string;
  paymentErrorMessage: string;
  postCodeLookUpAddresses: PostcodeLookupAddress[];
  deliveryDetails: DeliveryDetails;
}

export interface CustomerDetails {
  title?: string;
  firstName: string;
  lastName: string;
  line1: string;
  line2: string;
  city: string;
  postcode: string;
  email: string;
  phone: string;
  surgeryName?: string;
  surgeryAddress?: string;
  allowMarketing?: boolean;
}

export interface AddCustomerDetailsAction {
  type: string;
  payload: {
    customerDetails: CustomerDetails;
  };
}

export interface DeliveryDetails {
  title?: string;
  firstName: string;
  lastName: string;
  line1: string;
  line2: string;
  city: string;
  postcode: string;
}

export interface AddDeliveryDetailsAction {
  type: string;
  payload: {
    deliveryDetails: DeliveryDetails;
  };
}

export interface FetchBraintreeTokenAction {
  type: string;
  [RSAA]: {
    endpoint: string;
    method: string;
    types: [string, string, string];
  };
}

export interface FetchBraintreeTokenResponseAction {
  type: string;
  payload: {
    token: string;
  };
}

export interface PaymentFailureAction {
  type: string;
  payload: {
    response: {
      message: string | null;
    } | null;
  };
}

export interface FetchAddressFromPostcodeActionResponse {
  type: string;
  payload: {
    addresses: PostcodeLookupAddress[];
  };
}
