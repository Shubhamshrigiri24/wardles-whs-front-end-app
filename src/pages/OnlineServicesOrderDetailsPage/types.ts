import { SUBSCRIPTION_STATUSES } from "app/store/reducer/subscriptions/constants";
import { Discount } from "app/store/reducer/order/types";

export type OrderDetailProductInfo = {
  id: string;
  productName: string;
  activeIngredient: string;
  imageURI: string;
  worksIn: string;
  duration: string;
  isBestValue: boolean;
  canBuy: boolean;
  description: string;
  moreInfo: {
    title: string;
    content: string;
  }[];
  subscriptionMoreInfo?: string;
  levels: {
    levelLabel: string;
    columnLabelTemplate: string;
    pricePerUnitTemplate?: string;
    shouldDeriveBestValue?: boolean;
    stealPriceFromSelectedChild?: boolean;
    renderOrder: number;
  }[];
  priceTemplate: string;
  buyLinkURITemplate: string;
  startConsultationLinkURI: string;
};

export type OrderPricingType = Discount & {
  delivery?: { code: "PREMIUM" | "FREE"; value: number };
};

export type OrderDetailsProduct = {
  name: string;
  sku: string;
  label: string;
  itemsPerPack: number;
  price: number;
  pricePerUnit: number;
  isBestValue: boolean;
  meta: Record<string, any>;
  productInfo: OrderDetailProductInfo;
};

export type Subscription = {
  startDate: string;
  renewalDate: string;
  nextShipment: string;
  status: typeof SUBSCRIPTION_STATUSES;
};

export type Payment = {
  cardNumber: string;
  hasError: boolean;
};

export type Delivery = {
  firstName: string;
  lastName: string;
  postcode: string;
  addressline1: string;
  addressline2: string;
  city: string;
  email: string;
  phone: string;
  product: any;
  pricing: OrderPricingType;
  shouldShowDeliveryInfo?: boolean;
};

export enum OrderType {
  SUBSCRIPTION = "subscription",
  INDIVIDUAL_ORDER = "order",
}

export enum ConfirmationState {
  INITIAL = "initial",
  CONFIRMED = "confirmed",
}
