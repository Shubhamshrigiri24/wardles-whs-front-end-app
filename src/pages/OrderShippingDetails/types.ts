import { OrderCustomerDetails } from "app/store/reducer/order/types";

export type AddressValues = {
  line1: string;
  line2: string;
  city: string;
  postcode: string;
};

export type FormValues = OrderCustomerDetails & {
  agreeTerms: boolean;
};

export type PostcodeAddress = {
  city: string;
  country: string;
  postcode: string;
  line1: string;
  line2: string;
};

export type Scenarios = {
  isAuthenticatedAndHasAccount: boolean;
  isAuthenticatedAndRequestsAccount: boolean;
  isNotAuthenticatedAndRequestsAccount: boolean;
  isAuthenticatedAndHasRequestedAccount: boolean;
  isNotAuthenticatedAndDoesNotRequestAccount: boolean;
};

export enum FormStep {
  ACCOUNT_DETAILS = "ACCOUTN_DETAILS",
  DELIVERY_DETAILS = "DELIVERY_DETAILS",
}
