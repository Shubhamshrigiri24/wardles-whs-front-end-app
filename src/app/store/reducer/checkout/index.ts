import { handleActions } from "redux-actions";
import { AnyAction, combineReducers } from "redux";
import {
  CustomerDetails,
  AddCustomerDetailsAction,
  FetchBraintreeTokenResponseAction,
  FetchAddressFromPostcodeActionResponse,
  DeliveryDetails,
  AddDeliveryDetailsAction,
} from "app/store/reducer/checkout/types";
import {
  ADD_CUSTOMER_DETAILS,
  ADD_DELIVERY_DETAILS,
  FETCH_ADDRESS_FROM_POSTCODE_FAILURE,
  FETCH_ADDRESS_FROM_POSTCODE_RESPONSE,
  FETCH_BRAINTREE_TOKEN_REQUEST,
  FETCH_BRAINTREE_TOKEN_RESPONSE,
  emptyDeliveryDetails,
} from "app/store/reducer/checkout/constants";

import { PostcodeLookupAddress } from "../shared";

const defaultCustomerDetails: CustomerDetails = {
  firstName: "",
  lastName: "",
  email: "",
  postcode: "",
  line1: "",
  line2: "",
  city: "",
  phone: "",
  surgeryName: "",
  surgeryAddress: "",
  allowMarketing: false,
};

const customerDetails = handleActions(
  {
    [ADD_CUSTOMER_DETAILS]: (_state: CustomerDetails, action: AnyAction) => {
      const a = action as AddCustomerDetailsAction;
      return a.payload.customerDetails;
    },
  },
  defaultCustomerDetails
);

const deliveryDetails = handleActions(
  {
    [ADD_DELIVERY_DETAILS]: (
      _state: DeliveryDetails,
      action: AddDeliveryDetailsAction
    ) => {
      return action.payload.deliveryDetails;
    },
  },
  emptyDeliveryDetails
);

const braintreeToken = handleActions(
  {
    [ADD_CUSTOMER_DETAILS]: () => "",
    [FETCH_BRAINTREE_TOKEN_REQUEST]: () => "",
    [FETCH_BRAINTREE_TOKEN_RESPONSE]: (_state: string, action: AnyAction) => {
      const a = action as FetchBraintreeTokenResponseAction;
      return a.payload.token;
    },
  },
  ""
);

const paymentErrorMessage = handleActions(
  {
    [FETCH_BRAINTREE_TOKEN_REQUEST]: () => "",
  },
  ""
);

const confirmedPayment = handleActions(
  {
    [FETCH_BRAINTREE_TOKEN_REQUEST]: () => false,
    [ADD_CUSTOMER_DETAILS]: () => false,
  },
  false
);

const postCodeLookUpAddresses = handleActions(
  {
    [FETCH_ADDRESS_FROM_POSTCODE_RESPONSE]: (
      _state: PostcodeLookupAddress[],
      action: AnyAction
    ) => {
      const a = action as FetchAddressFromPostcodeActionResponse;
      return a.payload.addresses;
    },
    [FETCH_ADDRESS_FROM_POSTCODE_FAILURE]: () => [],
  },
  []
);

const reducer = combineReducers({
  customerDetails,
  confirmedPayment,
  braintreeToken,
  paymentErrorMessage,
  postCodeLookUpAddresses,
  deliveryDetails,
});

export default reducer;
