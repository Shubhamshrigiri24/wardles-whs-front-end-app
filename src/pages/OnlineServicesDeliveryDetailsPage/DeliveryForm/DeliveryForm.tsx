import React, { useCallback, useEffect, useState } from "react";
import {
  Typography,
  Button,
  TextField,
  Theme,
  makeStyles,
  FormControl,
  InputLabel,
  Select,
  MenuItem,
  Container,
  Card,
  Checkbox,
  FormHelperText,
} from "@material-ui/core";
import { useForm } from "react-hook-form";
import * as yup from "yup";
import {
  CustomerDetails,
  DeliveryDetails,
} from "app/store/reducer/checkout/types";
import { emptyDeliveryDetails } from "app/store/reducer/checkout/constants";
import { PostcodeLookupAddress } from "app/store/reducer/shared";
import AddressLookUp from "../../../components/AddressLookUp";
import { POSTCODE_PATTERN } from "../../../utils/regex/postcode";
import capitalize from "lodash/capitalize";

const Schema = yup.object().shape({
  firstName: yup.string().trim().required("Enter your first name"),
  lastName: yup.string().trim().required("Enter your last name"),
  line1: yup
    .string()
    .trim()
    .required("Enter the first line of your delivery address"),
  line2: yup.string().trim(),
  city: yup.string().trim().required("Enter a city"),
  postcode: yup
    .string()
    .trim()
    .required("Enter a valid postcode")
    .matches(POSTCODE_PATTERN, "Enter a valid postcode"),
  title: yup.string().trim().required("Enter your title"),
});

const useStyles = makeStyles((theme: Theme) => ({
  form: {
    margin: "0 auto",
    padding: theme.spacing(2),
    [theme.breakpoints.down("sm")]: {
      padding: theme.spacing(0),
    },
    border: `1px solid ${theme.palette.divider}`,
  },
  inputMargin: {
    margin: theme.spacing(2, 0),
  },
  paddingBottom: {
    paddingBottom: theme.spacing(2),
  },
  marginTopBottom: {
    marginBottom: theme.spacing(3),

    [theme.breakpoints.down("sm")]: {
      margin: theme.spacing(3, 0),
    },
  },
  marginTop: {
    marginTop: theme.spacing(4),
  },
  input: {
    marginTop: theme.spacing(2),
    width: "100%",
  },
  pageMargin: {
    marginTop: theme.spacing(2),
    marginBottom: theme.spacing(15),
    [theme.breakpoints.down("sm")]: {
      margin: theme.spacing(0),
    },
  },
  card: {
    border: "none",
    background: "none",
    [theme.breakpoints.down("sm")]: {
      margin: theme.spacing(0),
      padding: "0px",
    },
  },
  checkboxText: {
    color: "#65727D",
    marginBottom: theme.spacing(3),
  },
  textLink: {
    textDecoration: "none",
    color: "#0061F2",
  },
  hidden: { display: "none" },
}));

export interface DeliveryDetailsFormProps {
  onSubmit(data: DeliveryDetails): void;
  customerDetails: CustomerDetails;
  fetchAddressesByPostcode(postcode: string): void;
  postCodeLookUpAddresses: PostcodeLookupAddress[];
}

interface DeliveryDetailsFormData {
  title: string;
  firstName: string;
  lastName: string;
  line1: string;
  line2: string;
  city: string;
  postcode: string;
}

const DeliveryDetailsForm: React.FC<DeliveryDetailsFormProps> = (props) => {
  const {
    onSubmit,
    postCodeLookUpAddresses,
    fetchAddressesByPostcode,
    customerDetails,
  } = props;
  const classes = useStyles();
  // eslint-disable-next-line @typescript-eslint/naming-convention
  const [useCustomerDetails, setUseCustomerDetails] = useState(false);

  const {
    register,
    handleSubmit,
    errors,
    reset,
    setValue,
    triggerValidation,
    getValues,
  } = useForm({
    validationSchema: Schema,
    defaultValues: emptyDeliveryDetails,
    submitFocusError: false,
  });

  useEffect(() => {
    let initialDeliveryDetails = emptyDeliveryDetails;
    if (useCustomerDetails) {
      initialDeliveryDetails.title = customerDetails.title;
      initialDeliveryDetails.firstName = customerDetails.firstName;
      initialDeliveryDetails.lastName = customerDetails.lastName;
      initialDeliveryDetails.line1 = customerDetails.line1;
      initialDeliveryDetails.line2 = customerDetails.line2;
      initialDeliveryDetails.city = customerDetails.city;
      initialDeliveryDetails.postcode = customerDetails.postcode;
      reset(initialDeliveryDetails);
    } else {
      reset(emptyDeliveryDetails);
    }
  }, [customerDetails, reset, useCustomerDetails]);

  const handleTitleChange = useCallback(
    (
      event: React.ChangeEvent<{
        name?: string;
        value: unknown;
      }>
    ) => {
      setValue([{ title: event.target.value as string }]);
      triggerValidation("title");
    },
    [setValue, triggerValidation]
  );

  const setAddressFromAddressLookUp = useCallback(
    (address: PostcodeLookupAddress) => {
      setValue([
        { postcode: address.postcode },
        { line1: address.line1 },
        { line2: address.line2 },
        { city: address.city },
      ]);
      triggerValidation("postcode");
      triggerValidation("line1");
      triggerValidation("line2");
      triggerValidation("city");
    },
    [setValue, triggerValidation]
  );

  const DeliveryDetailsFormOnSubmit = useCallback(
    (data: DeliveryDetailsFormData) => {
      onSubmit({ ...data });
    },
    [onSubmit]
  );

  const checkUseCustomerDetails = (event: any) => {
    setUseCustomerDetails(event.target.checked);
  };

  const validateFormOnchange = useCallback(() => {}, []);

  return (
    <Container maxWidth={"sm"} className={classes.pageMargin}>
      <Card className={classes.card}>
        <Typography variant={"h3"} className={classes.marginTopBottom}>
          Delivery details
        </Typography>
        <Checkbox
          onChange={checkUseCustomerDetails}
          style={{
            paddingLeft: "0px",
          }}
          inputProps={{
            // @ts-ignore
            "data-testid": "DeliveryDetailsForm/checkbox",
          }}
        />
        <Typography variant={"caption"} className={classes.checkboxText}>
          Use same details ({capitalize(customerDetails.firstName)}{" "}
          {capitalize(customerDetails.lastName)},{" "}
          {customerDetails.postcode.toUpperCase()})
        </Typography>
        <form
          id={"DeliveryDetails-details-form"}
          noValidate
          autoComplete={"off"}
          onChange={handleSubmit(validateFormOnchange)}
          onSubmit={handleSubmit(DeliveryDetailsFormOnSubmit)}
        >
          <div className={classes.paddingBottom}>
            <FormControl
              variant={"outlined"}
              className={classes.input}
              error={!!errors.title && !!errors.title.message}
            >
              <InputLabel
                htmlFor={"outlined-title-select"}
                style={{ backgroundColor: "white" }}
              >
                Title
              </InputLabel>
              <Select
                key={"DeliveryDetailsForm/title-select"}
                value={getValues("title")}
                onChange={handleTitleChange}
                inputRef={register}
                inputProps={{
                  name: "title",
                  "data-testid": "DeliveryDetailsForm/title-select",
                  id: "outlined-title-select",
                }}
              >
                {["Mr", "Mrs", "Miss", "Ms", "Mx", "Sir", "Dr"].map(
                  (title, index) => {
                    return (
                      <MenuItem key={`${title}-${index}`} value={title}>
                        {title}
                      </MenuItem>
                    );
                  }
                )}
              </Select>
              {errors.title && errors.title.message && (
                <FormHelperText>{errors.title.message}</FormHelperText>
              )}
            </FormControl>

            <TextField
              name={"title"}
              inputRef={register}
              className={classes.hidden}
            />
          </div>

          <TextField
            variant={"outlined"}
            fullWidth
            label={"First name"}
            name={"firstName"}
            error={!!errors.firstName}
            helperText={errors.firstName && errors.firstName.message}
            inputRef={register}
            InputLabelProps={{
              shrink: true,
            }}
            inputProps={{
              "data-testid": "DeliveryDetailsForm/firstName",
            }}
          />

          <TextField
            className={classes.inputMargin}
            label={"Last name"}
            variant={"outlined"}
            fullWidth
            name={"lastName"}
            error={!!errors.lastName}
            helperText={errors.lastName && errors.lastName.message}
            inputRef={register}
            InputLabelProps={{
              shrink: true,
            }}
            inputProps={{
              "data-testid": "lastName",
            }}
          />

          <AddressLookUp
            postCodeLookUpAddresses={postCodeLookUpAddresses}
            fetchAddressesByPostcode={fetchAddressesByPostcode}
            setAddress={setAddressFromAddressLookUp}
            error={errors?.postcode?.message?.toString() || ""}
            inputRef={register}
          />

          <TextField
            className={classes.input}
            label={"Address line 1"}
            variant={"outlined"}
            fullWidth
            name={"line1"}
            error={!!errors.line1}
            helperText={errors.line1 && errors.line1.message}
            inputRef={register}
            InputLabelProps={{
              shrink: true,
            }}
            inputProps={{
              "data-testid": "line1",
            }}
          />
          <TextField
            className={classes.input}
            label={"Address line 2 (optional)"}
            variant={"outlined"}
            fullWidth
            name={"line2"}
            error={!!errors.line2}
            helperText={errors.line2 && errors.line2.message}
            inputRef={register}
            InputLabelProps={{
              shrink: true,
            }}
            inputProps={{
              "data-testid": "line2",
            }}
          />
          <TextField
            className={classes.input}
            label={"City"}
            variant={"outlined"}
            fullWidth
            name={"city"}
            error={!!errors.city}
            helperText={errors.city && errors.city.message}
            inputRef={register}
            InputLabelProps={{
              shrink: true,
            }}
            inputProps={{
              "data-testid": "DeliveryDetailsForm/city",
            }}
          />

          <Button
            type={"submit"}
            color={"primary"}
            variant={"contained"}
            data-testid={"DeliveryDetailsForm/next-button"}
            className={classes.input}
          >
            Next
          </Button>
        </form>
        <div className={classes.input}>
          <Typography variant={"caption"}>
            By entering your details and payment information, you are agreeing
            to our{" "}
            <a
              className={classes.textLink}
              href={
                "https://www.well.co.uk/about-us/policies/shop-terms-and-conditions"
              }
              target={"_blank"}
              rel={"noopener noreferrer"}
            >
              Terms and Conditions
            </a>
            .
          </Typography>
        </div>
      </Card>
    </Container>
  );
};

export default DeliveryDetailsForm;
