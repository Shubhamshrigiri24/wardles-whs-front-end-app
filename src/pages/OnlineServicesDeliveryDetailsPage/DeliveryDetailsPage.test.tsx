import React from "react";
import { render, fireEvent, act } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import DeliveryForm, {
  DeliveryDetailsFormProps,
} from "./DeliveryForm/DeliveryForm";
import { CustomerDetails } from "../../app/store/reducer/checkout/types";

import { makePostCodeLookUpAddresses } from "app/store/reducer/mock";

const mockHistoryPush = jest.fn();
jest.mock("react-router-dom", () => ({
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));
beforeEach(() => {
  mockHistoryPush.mockReset();
});

describe("<DeliveryForm />", () => {
  it("should match the snapshot", () => {
    const props = makeProps();
    const { container } = render(<DeliveryForm {...props} />);
    expect(container).toMatchSnapshot();
  });
  it("should auto fill form if 'use customer details' box is ticked", async () => {
    const props = makeProps();
    const { getByTestId } = render(<DeliveryForm {...props} />);
    const checkbox = getByTestId("DeliveryDetailsForm/checkbox");
    await act(async () => {
      fireEvent.click(checkbox);
    });

    expect(checkbox).toHaveProperty("checked", true);
    expect(getByTestId("DeliveryDetailsForm/firstName")).toHaveProperty(
      "value",
      "firstName"
    );
    expect(getByTestId("DeliveryDetailsForm/city")).toHaveProperty(
      "value",
      "delivery city"
    );
  });
  it.skip("should call onSubmit on form submit with delivery details", async () => {
    const props = makeProps();
    const { getByTestId } = render(<DeliveryForm {...props} />);

    const checkbox = getByTestId("DeliveryDetailsForm/checkbox");
    await act(async () => {
      fireEvent.click(checkbox);
    });

    await act(async () => {
      fireEvent.change(getByTestId("DeliveryDetailsForm/title-select"), {
        target: {
          value: "Miss",
        },
      });
    });

    await act(async () => {
      fireEvent.click(getByTestId("DeliveryDetailsForm/next-button"));
    });
    expect(props.onSubmit).toBeCalledWith({
      title: "Miss",
      firstName: "firstName",
      lastName: "lastName",
      postcode: "M19 2UL",
      line1: "line1",
      line2: "line2",
      city: "delivery city",
    });
  });
});

function makeProps(
  v?: Partial<DeliveryDetailsFormProps>
): DeliveryDetailsFormProps {
  return {
    onSubmit: jest.fn(),
    customerDetails: customerDetails,
    fetchAddressesByPostcode: jest.fn(),
    postCodeLookUpAddresses: makePostCodeLookUpAddresses(),
  };
}

const customerDetails: CustomerDetails = {
  firstName: "firstName",
  lastName: "lastName",
  email: "a@b.com",
  postcode: "M19 2UL",
  line1: "line1",
  line2: "line2",
  city: "delivery city",
  phone: "01189998819991197253",
  surgeryName: "surgeryName",
  surgeryAddress: "surgeryAddress",
  allowMarketing: true,
};
