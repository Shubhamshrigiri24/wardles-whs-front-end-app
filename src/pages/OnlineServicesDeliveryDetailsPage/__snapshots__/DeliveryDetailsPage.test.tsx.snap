// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeliveryForm /> should match the snapshot 1`] = `
<div>
  <div
    class="MuiContainer-root makeStyles-pageMargin-7 MuiContainer-maxWidthSm"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-8 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <h3
        class="MuiTypography-root makeStyles-marginTopBottom-4 MuiTypography-h3"
      >
        Delivery details
      </h3>
      <span
        aria-disabled="false"
        class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-12 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
        style="padding-left: 0px;"
      >
        <span
          class="MuiIconButton-label"
        >
          <input
            class="PrivateSwitchBase-input-15"
            data-indeterminate="false"
            data-testid="DeliveryDetailsForm/checkbox"
            type="checkbox"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </span>
      <span
        class="MuiTypography-root makeStyles-checkboxText-9 MuiTypography-caption"
      >
        Use same details (
        Firstname
         
        Lastname
        ,
         
        M19 2UL
        )
      </span>
      <form
        autocomplete="off"
        id="DeliveryDetails-details-form"
        novalidate=""
      >
        <div
          class="makeStyles-paddingBottom-3"
        >
          <div
            class="MuiFormControl-root makeStyles-input-6"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink="false"
              for="outlined-title-select"
              style="background-color: white;"
            >
              Title
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            >
              <div
                aria-haspopup="listbox"
                aria-labelledby="mui-component-select-title"
                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
                id="mui-component-select-title"
                role="button"
                tabindex="0"
              >
                <span>
                  ​
                </span>
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput"
                data-testid="DeliveryDetailsForm/title-select"
                id="outlined-title-select"
                name="title"
                tabindex="-1"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-17"
                  style="width: 0.01px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root makeStyles-hidden-11"
          >
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="title"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
            data-shrink="true"
          >
            First name
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              data-testid="DeliveryDetailsForm/firstName"
              name="firstName"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-18 PrivateNotchedOutline-legendNotched-19"
              >
                <span>
                  First name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-inputMargin-2 MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
            data-shrink="true"
          >
            Last name
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              data-testid="lastName"
              name="lastName"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-18 PrivateNotchedOutline-legendNotched-19"
              >
                <span>
                  Last name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-9"
          >
            <div
              class="MuiFormControl-root MuiTextField-root makeStyles-input-20"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
                data-shrink="true"
                for="postcode-input"
                id="postcode-input-label"
              >
                Postcode
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input"
                  data-testid="postcode"
                  id="postcode-input"
                  name="postcode"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="PrivateNotchedOutline-legendLabelled-18 PrivateNotchedOutline-legendNotched-19"
                  >
                    <span>
                      Postcode
                    </span>
                  </legend>
                </fieldset>
              </div>
              
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-searchButton-21 Mui-disabled Mui-disabled"
              data-testid="addressLookUp/search-button"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Search
              </span>
            </button>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-input-6 MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
            data-shrink="true"
          >
            Address line 1
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              data-testid="line1"
              name="line1"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-18 PrivateNotchedOutline-legendNotched-19"
              >
                <span>
                  Address line 1
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-input-6 MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
            data-shrink="true"
          >
            Address line 2 (optional)
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              data-testid="line2"
              name="line2"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-18 PrivateNotchedOutline-legendNotched-19"
              >
                <span>
                  Address line 2 (optional)
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-input-6 MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined"
            data-shrink="true"
          >
            City
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              data-testid="DeliveryDetailsForm/city"
              name="city"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-16 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-18 PrivateNotchedOutline-legendNotched-19"
              >
                <span>
                  City
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-input-6 MuiButton-containedPrimary"
          data-testid="DeliveryDetailsForm/next-button"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Next
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </form>
      <div
        class="makeStyles-input-6"
      >
        <span
          class="MuiTypography-root MuiTypography-caption"
        >
          By entering your details and payment information, you are agreeing to our
           
          <a
            class="makeStyles-textLink-10"
            href="https://www.well.co.uk/about-us/policies/shop-terms-and-conditions"
            rel="noopener noreferrer"
            target="_blank"
          >
            Terms and Conditions
          </a>
          .
        </span>
      </div>
    </div>
  </div>
</div>
`;
